### ISSUE TITLE
BE embed media helper refactor for video and audio

### ISSUE THREAD

**Issue by edoardocavazza** [bedita/bedita#562]
- [x] use HTML5 tags for Video and Audio elements
- [x] fallback using Flash and (opt) SIlverlight

see:
http://mediaelementjs.com/
http://www.videojs.com/

**Comment by stefanorosanelli** [comment]
A first implementation is present in __issue/562-media.helper-refactor__ branch
@edoardocavazza @batopa @xho can someone take a look ?

**Comment by edoardocavazza** [comment]
Something goes wrong if the param "URLonly" is passed as TRUE. It seems that it is not handled.

**Comment by batopa** [comment]
At the end I replaced mediaelement.js with video.js. mediaelement depends from jquery and in some frontends it could be an issue.

To test:

```
{$beEmbedMedia->object($object, $params, $attributes)}
```

where:
- `$object` is a video or audio object type
- `$params` in addition to the usual configurations can contain a key 'params' that will be placed in `<video data-setup="">` or `<audio data-setup="">` to configure video.js
- `$attributes` as usual are HTML attributes

For example

```
{$beEmbedMedia->object($object, [
        'params' => [        
            'controls' => false,
            'autoplay' => true
        ]
    ],
    [
        'width' => 500
    ]
)}
```

## Video

The width and height are calculated in this way:
- if no width and height are set then use the default in configuration (bedita.ini.php)
- if only width or height is set the other dimension is calculated to fit 16/9 ratio
- if both are defined use them

If no 'poster' is passed in `$attributes` and the object to embed has a poster relation then it will be used  else if it's defined `$object['thumbnail']` will use it. The poster relation is searched following the typical frontend structure of objects i.e. `$object['relations']['poster'].

The poster thumbnail of related object (poster relation) is get using as base thumb params

``` php
$posterParams = array(
    'presentation' => 'thumb',
    'mode' => 'fill',
    'upscale' => true,
    'URLonly' => true
);
```

width and height are set equal to that calculated above.

## Audio

If width and height aren't passed then set the width to default in configuration. The height is handled by video.js. 

The poster relation is handled as in video.

## id

The id for `<video>` and `<audio>` is in the form of "video_nickname-object" and "audio_nickname-object" so  to access to relative video.js object from js:

``` javascript
var v = videojs('video_nickname-object');
v.play();
...
```

If the same object is embedded many times in the same page then an hash is appended to id to avoid id collision.

At the moment lacks the ability to customize the css. Now `video-js.min.css` is always included. We could use it in backend while in frontend app we could check the existence of `webroot/css/video-js.css` file and fallback to backend css in case.

@qwerg @edoardocavazza @xho any thoughts?

**Comment by fquffio** [comment]
Tested and merged to 3-corylus with pull request #738. @batopa If everything still looks alright, I think we could close this issue.

**Comment by batopa** [comment]
@fquffio it seems all ok! Closing
